clang/test/Analysis/issue-151529.cpp

```diff
@@ -0,0 +1,15 @@
+// RUN: %clang_analyze_cc1 -std=c++23 -analyzer-checker=cplusplus -verify %s
+// expected-no-diagnostics
```

Comment:
Could you move this test to `clang/test/Analysis/builtin_assume.cpp`?
Its so small that I don't think it deserves being separate.